### ISSUE TITLE
Unclear documentation for the Github status reporter

### ISSUE THREAD

**Issue by levitte** [buildbot/buildbot#6504]
The GitHubStatusPush documentation indicates that a Github OAuth App needs to be created.

Fine, I can do that, following https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app

But, there's that mandatory field, "Authorization Callback URL"...  I have *no* idea what to fill in there.

Help?

**Comment by yan12125** [comment]
For me a [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) also works fine with `GitHubStatusPush`. Documentation may be improved to mention that.

**Comment by levitte** [comment]
> For me a [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) also works fine with `GitHubStatusPush`. Documentation may be improved to mention that.

I've heard of that, and yeah, that works for status reporting on PRs. I was hoping to integrate master builds as well, the same way you can with github actions, and that would require having it as an actual github app (oauth or not), would it not?

Maybe I'm asking for too much...

**Comment by yan12125** [comment]
FWIW, [my setup](https://gitlab.com/yan12125/buildbot/-/blob/master/master.cfg#L110) is primarily for reporting status to the master branch. This is how it looks like (sorry for a screenshot instead of a link - it's a private repo)

![圖片](https://user-images.githubusercontent.com/1937689/167245797-d6979b79-a845-4b73-bd35-645ffa62cf3e.png)

Those red crosses and green checks besides commits are quite similar to ones on https://github.com/buildbot/buildbot/commits/master. I didn't find configuration for buildbot itself, though.


> Maybe I'm asking for too much...

Not at all :)

**Comment by levitte** [comment]
> FWIW, [my setup](https://gitlab.com/yan12125/buildbot/-/blob/master/master.cfg#L110) is primarily for reporting status to the master branch. This is how it looks like (sorry for a screenshot instead of a link - it's a private repo)

Oh *that's* where you find them!!!  I was expecting to see a "Checks" tab in https://<hostname>/<owner>/<repo>, it took me a moment to figure that I had to go to the commit list to see the build status markers...
That wasn't entirely obvious!  (yeah, looking at you, Github! :wink:)

All right, thanks for that hint, that was the last bit I needed to get a clear picture.

**Comment by yan12125** [comment]
Yeah, seems there is no "Check" tab like PRs. Status in the commit list is not very good user interface IMO. Usually I use buildbot's grid view or console view to check status for each commit.

**Comment by levitte** [comment]
> Yeah, seems there is no "Check" tab like PRs. Status in the commit list is not very good user interface IMO. Usually I use buildbot's grid view or console view to check status for each commit.

Yeah, same here...  but my users like to do things within github where possible.
Ah well, it's really just a matter of instructing them where to look, and it's really not that bad once you get the hang of it.

But, back to the issue at hand, it seems like having buildbot as an OAuth app has stopped working, 'cause github.
The documentation needs to be changed accordingly...
